devices/System.Drawing/System.Drawing.cs

```diff
@@ -180,9 +180,82 @@ public float GetSaturation()
         }
 
         /// <Summary>Gets the 32-bit ARGB value of this System.Drawing.Color structure.</Summary>
-        /// <returns>The 32-bit ARGB value of this System.Drawing.Color.</returns>     
+        /// <returns>The 32-bit ARGB value of this System.Drawing.Color.</returns>
         public int ToArgb() => (int)_color;
 
+        /// <summary>
+        /// Internal Helper function for ParseHex
+        /// </summary>
+        private static int ParseHexChar(char c)
+        {
+            const int zeroChar = (int)'0';
+            const int aLowerChar = (int)'a';
+            const int aUpperChar = (int)'A';
+            var intChar = (int)c;
+
+            return intChar switch
+            {
+                >= zeroChar and <= zeroChar + 9 => (intChar - zeroChar),
+                >= aLowerChar and <= aLowerChar + 5 => (intChar - aLowerChar + 10),
+                >= aUpperChar and <= aUpperChar + 5 => (intChar - aUpperChar + 10),
+                _ => throw new FormatException($"Illegal token. {c} can't be converted")
+            };
+        }
+
+        /// <summary>
+        /// Convert String into an Color struct.
+        /// </summary>
+        /// <param name="hexString">Color String. Allowed formats are #AARRGGBB #RRGGBB #ARGB #RGB</param>
+        /// <returns>returns an Color struct otherwise throws an exception</returns>
+        /// <exception>ArgumentException or FormatException</exception>
+        public static Color ParseHex(string hexString)
+        {
+            int r, g, b;
+            var a = 255;
+            if (hexString[0] != '#')
+                throw new ArgumentException("Leading # is missing.");
+            switch (hexString.Length)
+            {
+                case 9: // #AARRGGBB
+                    a = ParseHexChar(hexString[1]) * 16 + ParseHexChar(hexString[2]);
```

Comment:
This is easier to understand and also a bit faster:
```suggestion
                    a = ParseHexChar(hexString[1]) << 4 | ParseHexChar(hexString[2]);
```